### ISSUE TITLE
Statement parsing algorithm should be change

### ISSUE THREAD

**Issue by key262yek** [RustPython/RustPython#3882]
## Feature

In CPython grammar (Grammar/python.gram) line 695, there are group expression type exists.
```gram
atom[expr_ty]:
    ...
     | &'(' (tuple | group | genexp)
```

The definition of group is at line 709 of same file
```gram
group[expr_ty]:
    | '(' a=(yield_expr | named_expression) ')' { a }
    | invalid_group
```
and of invalid group is at line 947
```gram
invalid_group:
    | '(' a=starred_expression ')' {
        RAISE_SYNTAX_ERROR_KNOWN_LOCATION(a, "cannot use starred expression here") }
    | '(' a='**' expression ')' {
        RAISE_SYNTAX_ERROR_KNOWN_LOCATION(a, "cannot use double starred expression here") }
```

Here, the first pattern in invalid_group is related to SyntaxError of `del (*x)` (pr #3879)
It should print **cannot use starred expression here**, but without group expression it makes **cannot delete starred**.

In parser/src/python.lalrpop at line 956, only tuple, yield, genexp are parsed. 
Even, parenthesized-form is missing. 

I think, we need to make group expression to parse parenthesized-form and handle error about starred epression,
 or add equivalent parsing grammar into atom parser

## Python Documentation

Maybe those are related things. 
[atom](https://docs.python.org/3/reference/expressions.html#atoms)
[parenthesized-forms](https://docs.python.org/3/reference/expressions.html#parenthesized-forms)

**Comment by key262yek** [comment]
Since `group` is not actual expression kind, it is better to find equivalent parsing method for both parenthesized forms and starred expression

**Comment by key262yek** [comment]
I think it is my mistake.
followings are all fine, so parenth_form and starred expression works. 
Hence it's just about error in del statement. 

```python
>>> x = [1, 2, 3]
>>> ([*x])
[1, 2, 3]
>>> (*x)
SyntaxError: cannot use starred expression here at line 1 column 2
(*x)
 ^
```

**Comment by key262yek** [comment]
I re-change my opinion. 
It is right that  missing a group expression is a problem.

In CPython, the expression `(*x)` raises SyntaxError in statement parsing level. 
However, in RustPython, we have no expression to handle it. 

In CPython,
```python3
>>> while (*x)
  File "<stdin>", line 1
    while (*x)
           ^^
SyntaxError: cannot use starred expression here
>>> if (*x)
  File "<stdin>", line 1
    if (*x)
        ^^
SyntaxError: cannot use starred expression here
```

But, in RustPython
```python3
>>>>> while (*x)
SyntaxError: invalid syntax. Got unexpected token Newline at line 1 column 11
while (*x)
          ^
>>>>> if (*x)
SyntaxError: invalid syntax. Got unexpected token Newline at line 1 column 8
if (*x)
       ^
```
because there is no such rule to parse it.

**Comment by key262yek** [comment]
It is resolved by pr #3879

**Comment by youknowone** [comment]
we usually manage it opposite way. adding `fix #3882` to the PR make it automatically close this issue when it is merged.

**Comment by key262yek** [comment]
@youknowone  Thanks for notice. then I need to reopen another one the pr will fix
